src/Compilers/CSharp/Portable/Utilities/TypeSymbolExtensions.cs

```diff
@@ -43,6 +43,17 @@ public static int CustomModifierCount(this TypeSymbol type)
                         TypeWithAnnotations pointedAtType = pointer.PointedAtTypeWithAnnotations;
                         return pointedAtType.CustomModifiers.Length + pointedAtType.Type.CustomModifierCount();
                     }
+                case SymbolKind.FunctionPointer:
+                    {
+                        var funcPtr = (FunctionPointerTypeSymbol)type;
+                        var count = funcPtr.Signature.ReturnTypeWithAnnotations.CustomModifiers.Length + funcPtr.Signature.ReturnType.CustomModifierCount();
+                        foreach (var param in funcPtr.Signature.Parameters)
+                        {
+                            count += param.TypeWithAnnotations.CustomModifiers.Length + param.Type.CustomModifierCount();
```

Comment:
>param.TypeWithAnnotations.CustomModifiers.Length + param.Type.CustomModifierCount() [](start = 37, length = 83)

We have this same pattern in each of the cases. Please consider extracting a `static int getCustomModifierCount(TypeWithAnnotations)` method.

Reply:
I'll just bring the next commit from my planned follow-up pr forward which does more here.

---
In reply to: [422381885](https://github.com/dotnet/roslyn/pull/44078#discussion_r422381885) [](ancestors = 422381885)